kats/src/main/kotlin/kats/Reader.kt

```diff
@@ -0,0 +1,16 @@
+package kats
+
+/**
+ * Basic implementation of the Reader monad. Provides an "implicit" context (configuration) for
+ * function execution. Intended to provide Dependency Injection.
+ */
+class Reader<C, A>(val rd: (C) -> A) {
```

Comment:
* Fixing `A` to declare it as `out` as it's definitely a covariant type.
* Type `C` **cannot** be contravariant (`in`), as it's occurring also in an `out` position in the `flatMap()` method argument: `crossinline f: (A) -> Reader<C, B>`
* Fixing `A`, `C` and `B` types to make them extend from `Any`. (Would be cool to have an explanation of why exactly do you need this top type bound to be imposed here as I don't understand it completely.